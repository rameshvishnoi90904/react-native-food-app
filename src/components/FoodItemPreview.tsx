import React from "react"
import { View, Text, StyleSheet, Image, Dimensions, TouchableOpacity, GestureResponderEvent } from "react-native"

import { CartItem, FoodItem } from "../utils/types"
const {width, height} = Dimensions.get('screen')
const FoodItemPreview = ({item, cartItem, navigateToProductDetail, addToCart, sectionDisabled, removeFromCart}: {item: FoodItem, navigateToProductDetail: (pItem: FoodItem) => void, addToCart: (pItem: FoodItem) => void, removeFromCart: (pItem: FoodItem) => void, cartItem: CartItem, sectionDisabled: string}) => {
    let actionButton;
    const soldOut = item.itemStock.quantityLeft == 0;
    if (!soldOut) {
        if (cartItem && cartItem.quantity > 0) {
            actionButton = (
                <View style={styles.cardFunctions}>
                   <TouchableOpacity onPress={() => removeFromCart(item)}>
                        <Text style={styles.cartAction}>-</Text>
                    </TouchableOpacity>
                    <Text style={styles.cartQuantity}>{cartItem.quantity}</Text>
                    <TouchableOpacity onPress={() => addToCart(item)}>
                        <Text style={styles.cartAction}>+</Text>
                    </TouchableOpacity>
                </View>
            )
        } else if (sectionDisabled) {
            actionButton = (
                <TouchableOpacity style={styles.addButton} onPress={() => {}}>
                    <Text style={styles.addText}>Not available</Text>
                </TouchableOpacity>
            )
        } else {
            actionButton = (
                <TouchableOpacity style={styles.addButton} onPress={() => addToCart(item)}>
                    <Text style={styles.addText}>Add</Text>
                </TouchableOpacity>
            )
        }
        
    } else {
        actionButton = (
            <TouchableOpacity style={styles.addButton} onPress={() => {}}>
                <Text style={styles.addText}>Sold out</Text>
            </TouchableOpacity>
        )
    }

    const price = item.unitPriceFractional/100;
    return (
        <TouchableOpacity style={styles.item} onPress={() => navigateToProductDetail(item)}>
            <Image src={item.imageUrl} style={styles.imagePreview}/>
            <View style={styles.itemTextContainer}>
                <Text style={styles.title} numberOfLines={2}>{item.label}</Text>
                <Text style={styles.description}numberOfLines={3}>{item.description}</Text>
                <Text style={styles.price}>{item.currency} {price}</Text>
            </View>
            {actionButton}
            {
                (sectionDisabled || soldOut) ? 
                    <View style={styles.disabledSection}>
                    </View>
                : null
            }
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    item: {
        marginBottom: 10,
        marginHorizontal: 5,
        borderWidth: 1,
        borderColor: '#fff',
        width: width/2 - 20,
        height: 350,
        borderRadius: 4,
        flexDirection: 'column',
        backgroundColor: 'white',
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 1,
        },
        shadowOpacity: 0.20,
        shadowRadius: 1.41,
        elevation: 2,
    },
    disabledSection: {
        backgroundColor: 'rgba(265,265,265,0.4)',
        position: 'absolute',
        width: width/2 - 20,
        height: 350,
        top: 0
    },
    imagePreview: {
        width: width/2 - 20 - 2,
        height: width/2 - 40 - 2,
        resizeMode: 'contain',
        backgroundColor: '#fff',
        borderTopRightRadius: 4,
        borderTopLeftRadius: 4,
    },
    itemTextContainer: {
        paddingHorizontal: 12,
        paddingTop: 15,
        height: 60,
        flex: 1,
    },
    title: {
        fontSize: 16,
        marginBottom: 6,
    },
    description: {
        fontSize: 12,
        lineHeight: 19,
        flex: 1
    },
    price: {
        marginVertical: 12,
    },
    addButton: {
        backgroundColor: 'rgb(250, 85, 83)',
        alignItems: 'center',
        height: 40,
        justifyContent: 'center',
        borderBottomRightRadius: 4,
        borderBottomLeftRadius: 4
    },
    addText: {
        color: 'white',
        fontWeight: '500'
    },
    cardFunctions: {
       borderWidth: 1,
       flexDirection: 'row' ,
       height: 40, 
       alignItems: 'center',
        borderBottomLeftRadius: 4,
        borderBottomRightRadius: 4
    },
    cartAction: {
        fontSize: 18,
        padding: 10
    },
    cartQuantity: {
        flex: 1,
        fontSize: 20,
        textAlign: 'center'
    }

})
export default FoodItemPreview;