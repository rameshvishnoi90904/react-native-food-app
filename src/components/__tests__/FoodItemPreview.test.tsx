import 'react-native';
import React from 'react';
import { render, act, fireEvent } from '@testing-library/react-native';
import {it, jest, expect} from '@jest/globals';
import FoodItemPreview from '../FoodItemPreview';

it('renders FoodItemPreview correctly', () => {
    const item = {
        "id": 116,
        "label": "Rasa Sayang Burg",
        "description": "Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.<br/><br/>Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.<br/><br/>*Patty is NOT allium-free. You can opt for allium-free patty. <br/>*Option to remove onions.",
        "displayOrder": 0,
        "unitPriceFractional": 1390,
        "currency": "SGD",
        "imageUrl": "https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt",
        "itemStock": {
            "quantityLeft": 999
        }
    }
    const navigateToProductDetail = jest.fn();
    const addToCart = jest.fn();
    const removeFromCart = jest.fn();

    const tree = render(<FoodItemPreview 
        navigateToProductDetail={navigateToProductDetail} 
        addToCart={addToCart} 
        removeFromCart={removeFromCart}
        sectionDisabled={false}
        cartItem={{foodId: item.id, quantity: 0}}
        item={item}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders FoodItemPreview for sold out', () => {
    const item = {
        "id": 116,
        "label": "Rasa Sayang Burg",
        "description": "Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.<br/><br/>Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.<br/><br/>*Patty is NOT allium-free. You can opt for allium-free patty. <br/>*Option to remove onions.",
        "displayOrder": 0,
        "unitPriceFractional": 1390,
        "currency": "SGD",
        "imageUrl": "https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt",
        "itemStock": {
            "quantityLeft": 0
        }
    }
    const navigateToProductDetail = jest.fn();
    const addToCart = jest.fn();
    const removeFromCart = jest.fn();

    const tree = render(<FoodItemPreview 
        navigateToProductDetail={navigateToProductDetail} 
        addToCart={addToCart} 
        removeFromCart={removeFromCart}
        sectionDisabled={false}
        cartItem={{foodId: item.id, quantity: 0}}
        item={item}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})

it('renders FoodItemPreview for sectionDisabled', () => {
    const item = {
        "id": 116,
        "label": "Rasa Sayang Burg",
        "description": "Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.<br/><br/>Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.<br/><br/>*Patty is NOT allium-free. You can opt for allium-free patty. <br/>*Option to remove onions.",
        "displayOrder": 0,
        "unitPriceFractional": 1390,
        "currency": "SGD",
        "imageUrl": "https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt",
        "itemStock": {
            "quantityLeft": 999
        }
    }
    const navigateToProductDetail = jest.fn();
    const addToCart = jest.fn();
    const removeFromCart = jest.fn();

    const tree = render(<FoodItemPreview 
        navigateToProductDetail={navigateToProductDetail} 
        addToCart={addToCart} 
        removeFromCart={removeFromCart}
        sectionDisabled={true}
        cartItem={{foodId: item.id, quantity: 0}}
        item={item}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})


it('renders FoodItemPreview and add item into cart ', async () => {
    const item = {
        "id": 116,
        "label": "Rasa Sayang Burg",
        "description": "Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.<br/><br/>Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.<br/><br/>*Patty is NOT allium-free. You can opt for allium-free patty. <br/>*Option to remove onions.",
        "displayOrder": 0,
        "unitPriceFractional": 1390,
        "currency": "SGD",
        "imageUrl": "https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt",
        "itemStock": {
            "quantityLeft": 999
        }
    }
    const navigateToProductDetail = jest.fn();
    const addToCart = jest.fn().mockImplementationOnce(() => {})
    const removeFromCart = jest.fn();

    const screen = render(<FoodItemPreview 
        navigateToProductDetail={navigateToProductDetail} 
        addToCart={addToCart} 
        removeFromCart={removeFromCart}
        sectionDisabled={false}
        cartItem={{foodId: item.id, quantity: 0}}
        item={item}
        />);

    

    const addCartButton = await screen.findByTestId(`add-cart-button-${item.id}`)
    expect(addCartButton).toBeTruthy()

    act(() => {
        fireEvent.press(addCartButton);
    })

    expect(addToCart).toBeCalledTimes(1);
})


it('renders FoodItemPreview and add item into cart ', async () => {
    const item = {
        "id": 116,
        "label": "Rasa Sayang Burg",
        "description": "Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.<br/><br/>Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.<br/><br/>*Patty is NOT allium-free. You can opt for allium-free patty. <br/>*Option to remove onions.",
        "displayOrder": 0,
        "unitPriceFractional": 1390,
        "currency": "SGD",
        "imageUrl": "https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt",
        "itemStock": {
            "quantityLeft": 999
        }
    }
    const navigateToProductDetail = jest.fn();
    const addToCart = jest.fn();
    const removeFromCart = jest.fn();

    const screen = render(<FoodItemPreview 
        navigateToProductDetail={navigateToProductDetail} 
        addToCart={addToCart} 
        removeFromCart={removeFromCart}
        sectionDisabled={false}
        cartItem={{foodId: item.id, quantity: 1}}
        item={item}
        />);

    
    const addCartButton = await screen.findByTestId(`add-cart-${item.id}`)
    expect(addCartButton).toBeTruthy()

    act(() => {
        fireEvent.press(addCartButton);
        fireEvent.press(addCartButton);
        fireEvent.press(addCartButton);
    })

    expect(addToCart).toBeCalledTimes(3);

    const removeFromCartButton = await screen.findByTestId(`remove-cart-${item.id}`)
    expect(removeFromCartButton).toBeTruthy()

    act(() => {
        fireEvent.press(removeFromCartButton);
        fireEvent.press(removeFromCartButton);
    })

    expect(removeFromCart).toBeCalledTimes(2);

    const cardItem = await screen.findByTestId(`food-card-${item.id}`)
    expect(removeFromCartButton).toBeTruthy()

    act(() => {
        fireEvent.press(cardItem);
    })

    expect(navigateToProductDetail).toBeCalledTimes(1)

})