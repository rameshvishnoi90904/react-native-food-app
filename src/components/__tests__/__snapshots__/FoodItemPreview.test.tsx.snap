// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FoodItemPreview correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderColor": "#fff",
      "borderRadius": 4,
      "borderWidth": 1,
      "elevation": 2,
      "flexDirection": "column",
      "height": 350,
      "marginBottom": 10,
      "marginHorizontal": 5,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
      "width": 355,
    }
  }
  testID="food-card-116"
>
  <Image
    src="https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt"
    style={
      {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "height": 333,
        "resizeMode": "contain",
        "width": 353,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "height": 60,
        "paddingHorizontal": 12,
        "paddingTop": 15,
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        {
          "fontSize": 16,
          "marginBottom": 6,
        }
      }
    >
      Rasa Sayang Burg
    </Text>
    <Text
      numberOfLines={3}
      style={
        {
          "flex": 1,
          "fontSize": 12,
          "lineHeight": 19,
        }
      }
    >
      Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.&lt;br/&gt;&lt;br/&gt;Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.&lt;br/&gt;&lt;br/&gt;*Patty is NOT allium-free. You can opt for allium-free patty. &lt;br/&gt;*Option to remove onions.
    </Text>
    <Text
      style={
        {
          "marginVertical": 12,
        }
      }
    >
      SGD
       
      13.9
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(250, 85, 83)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="add-cart-button-116"
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "500",
        }
      }
    >
      Add
    </Text>
  </View>
</View>
`;

exports[`renders FoodItemPreview for sectionDisabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderColor": "#fff",
      "borderRadius": 4,
      "borderWidth": 1,
      "elevation": 2,
      "flexDirection": "column",
      "height": 350,
      "marginBottom": 10,
      "marginHorizontal": 5,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
      "width": 355,
    }
  }
  testID="food-card-116"
>
  <Image
    src="https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt"
    style={
      {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "height": 333,
        "resizeMode": "contain",
        "width": 353,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "height": 60,
        "paddingHorizontal": 12,
        "paddingTop": 15,
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        {
          "fontSize": 16,
          "marginBottom": 6,
        }
      }
    >
      Rasa Sayang Burg
    </Text>
    <Text
      numberOfLines={3}
      style={
        {
          "flex": 1,
          "fontSize": 12,
          "lineHeight": 19,
        }
      }
    >
      Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.&lt;br/&gt;&lt;br/&gt;Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.&lt;br/&gt;&lt;br/&gt;*Patty is NOT allium-free. You can opt for allium-free patty. &lt;br/&gt;*Option to remove onions.
    </Text>
    <Text
      style={
        {
          "marginVertical": 12,
        }
      }
    >
      SGD
       
      13.9
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(250, 85, 83)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "500",
        }
      }
    >
      Not available
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "rgba(265,265,265,0.4)",
        "height": 350,
        "position": "absolute",
        "top": 0,
        "width": 355,
      }
    }
  />
</View>
`;

exports[`renders FoodItemPreview for sold out 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderColor": "#fff",
      "borderRadius": 4,
      "borderWidth": 1,
      "elevation": 2,
      "flexDirection": "column",
      "height": 350,
      "marginBottom": 10,
      "marginHorizontal": 5,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
      "width": 355,
    }
  }
  testID="food-card-116"
>
  <Image
    src="https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt"
    style={
      {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "height": 333,
        "resizeMode": "contain",
        "width": 353,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "height": 60,
        "paddingHorizontal": 12,
        "paddingTop": 15,
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        {
          "fontSize": 16,
          "marginBottom": 6,
        }
      }
    >
      Rasa Sayang Burg
    </Text>
    <Text
      numberOfLines={3}
      style={
        {
          "flex": 1,
          "fontSize": 12,
          "lineHeight": 19,
        }
      }
    >
      Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.&lt;br/&gt;&lt;br/&gt;Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.&lt;br/&gt;&lt;br/&gt;*Patty is NOT allium-free. You can opt for allium-free patty. &lt;br/&gt;*Option to remove onions.
    </Text>
    <Text
      style={
        {
          "marginVertical": 12,
        }
      }
    >
      SGD
       
      13.9
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(250, 85, 83)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "500",
        }
      }
    >
      Sold out
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "rgba(265,265,265,0.4)",
        "height": 350,
        "position": "absolute",
        "top": 0,
        "width": 355,
      }
    }
  />
</View>
`;
